src/components/sidebar/SafeListContextMenu/MultiAccountContextMenu.tsx

```diff
@@ -0,0 +1,107 @@
+import type { MouseEvent } from 'react'
+import { useState, type ReactElement } from 'react'
+import ListItemIcon from '@mui/material/ListItemIcon'
+import IconButton from '@mui/material/IconButton'
+import MoreVertIcon from '@mui/icons-material/MoreVert'
+import MenuItem from '@mui/material/MenuItem'
+import ListItemText from '@mui/material/ListItemText'
+
+import EntryDialog from '@/components/address-book/EntryDialog'
+import EditIcon from '@/public/images/common/edit.svg'
+import PlusIcon from '@/public/images/common/plus.svg'
+import ContextMenu from '@/components/common/ContextMenu'
+import { trackEvent, OVERVIEW_EVENTS, OVERVIEW_LABELS } from '@/services/analytics'
+import { SvgIcon } from '@mui/material'
+import { AppRoutes } from '@/config/routes'
+import router from 'next/router'
+import { CreateSafeOnNewChain } from '@/features/multichain/components/CreateSafeOnNewChain'
+
+enum ModalType {
+  RENAME = 'rename',
+  ADD_CHAIN = 'add_chain',
+}
+
+const defaultOpen = { [ModalType.RENAME]: false, [ModalType.ADD_CHAIN]: false }
+
+const MultiAccountContextMenu = ({
+  name,
+  address,
+  chainIds,
+}: {
+  name: string
+  address: string
+  chainIds: string[]
+}): ReactElement => {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | undefined>()
+  const [open, setOpen] = useState<typeof defaultOpen>(defaultOpen)
+
+  const trackingLabel =
+    router.pathname === AppRoutes.welcome.accounts ? OVERVIEW_LABELS.login_page : OVERVIEW_LABELS.sidebar
+
+  const handleOpenContextMenu = (e: MouseEvent<HTMLButtonElement, globalThis.MouseEvent>) => {
+    e.stopPropagation()
+    setAnchorEl(e.currentTarget)
+  }
+
+  const handleCloseContextMenu = (event: MouseEvent) => {
+    event.stopPropagation()
+    setAnchorEl(undefined)
+  }
+
+  const handleOpenModal =
+    (type: keyof typeof open, event: typeof OVERVIEW_EVENTS.SIDEBAR_RENAME | typeof OVERVIEW_EVENTS.SIDEBAR_RENAME) =>
```

Comment:
```suggestion
    (type: ModalType, event: typeof OVERVIEW_EVENTS.ADD_NEW_NETWORK | typeof OVERVIEW_EVENTS.SIDEBAR_RENAME) =>
```

Can we use the enum directly for the type?